Support rules_swift debug info remapping feature

jerrymarino opened the pull request:
This bumps tulsi to use the debug info prefix mapping setup by
`rules_swift` ( previously, we used another path for Pinterest internal
toolchain setup ).

By default, the variable is not set `HAMMER_USE_DEBUG_INFO_REMAPPING`
and is added to the README. In the future, this may be tured on, but is
left off if people want to use older swift releases

rahul-malik commented:
any concerns about introducing an lldbinit now and interacting with local lldbinit files? we’ve managed to avoid this for a while but maybe we can’t really do the remapping stuff we did with objc?

jerrymarino commented:
Hey @rahul-malik thanks for the review. Basically, to use remapping with swift and toolchain easily, we need an identical source map for swift. We've been using a custom mapping ` bazel-out/.... -> SHARED_CACHEABLE_PINTEREST_DIRECTORY`   and XCHammer writes an lldbinit to map to `SHARED_CACHEABLE_PINTEREST_DIRECTORY -> SRCROOT`. This has worked 100% perfect since adopting out Bazel without swift! Basically, in this PR I changed out `SHARED_CACHEABLE_PINTEREST_DIRECTORY` to use `.` to be aligned with `rules_swift`. 

Without this change, the problem is that `SHARED_CACHEABLE_PINTEREST_DIRECTORY` doesn't exist in `rules_swift` and I'd rather try to make it work without forking. I also updated the readme to describe how to use the CC rules without our custom toolchain.

PS: I think we can cleanup some logic in our custom toolchain to be more aligned with the default Bazel toolchain as they added the debug info mapping 👍
